nodebuilder/share/opts.go

```diff
@@ -11,14 +13,22 @@ import (
 
 // WithPeerManagerMetrics is a utility function to turn on peer manager metrics and that is
 // expected to be "invoked" by the fx lifecycle.
-func WithPeerManagerMetrics(m *peers.Manager) error {
-	return m.WithMetrics()
+func WithPeerManagerMetrics(managers []*peers.Manager) error {
+	var err error
+	for _, m := range managers {
+		err = errors.Join(err, m.WithMetrics())
```

Comment:
Nit: looks like this has `O(N**2)` complexity (see https://cs.opensource.google/go/go/+/refs/tags/go1.22.3:src/errors/join.go;l=19)

Not sure how important it's but collecting errors and than joining them looks simpler. Same in the `func` below and in `share/p2p/discovery/manager.go`